src/engine/renderer/tr_image.cpp

```diff
@@ -2223,13 +2244,73 @@ image_t *R_FindCubeImage( const char *imageName, imageParams_t &imageParams )
 				}
 			}
 
-			Log::Debug( "found %s multifile cube map '%s'", format.name, imageName );
-			image = R_CreateCubeImage( ( char * ) buffer, ( const byte ** ) pic, greatestEdge, greatestEdge, imageParams );
-			R_FreeCubePics( pic, i );
-			return image;
+			Log::Debug( "Found %d×%d×6 %s multifile cube map '%s'", greatestEdge, greatestEdge, format.name, imageName );
+			image_t *image = R_CreateCubeImage( imageName, ( const byte ** ) pic, greatestEdge, greatestEdge, imageParams );
+
+			R_FreeCubePics( pic, 6 );
+
+			if ( image )
+			{
+				return image;
+			}
 		}
+	}
+
+	// Look for 6-faces single-file CRN or KTX skybox.
+	{
+		char fileName[ 1024 ];
+		const char *imageName_p = &imageName[ 0 ];
+		const char *token = COM_ParseExt2( &imageName_p, false );
+		Q_strncpyz( fileName, token, sizeof( fileName ) );
+		const char *ext = COM_GetExtension( fileName );
 
-		R_FreeCubePics( pic, i );
+		// The Daemon's default strategy is to use the hardcoded path if it exists.
+		if ( *ext )
+		{
+			// Look for the correct loader and load the image is the file is found.
+			for ( const auto &loader : imageLoaders )
+			{
+				if ( !loader.cubemap ) continue;
+
+				if ( !Q_stricmp( ext, loader.ext ) )
+				{
+					/* Do not complain on missing file if the extension is hardcoded to a wrong one since
+					the file can exist with another extension and it will tested right after that. */
+					if ( FS::PakPath::FileExists( fileName ) )
+					{
+						image_t *image = R_LoadAndCreateCubeImage( imageName, fileName, &loader, imageParams );
+
+						if ( image )
+						{
+							return image;
+						}
+					}
+
+					// We have to break because the expected loader was found.
+					break;
+				}
+			}
+		}
+
+		/* If the image path is “some.thing” for a file named “some.thing.crn”,
+		first try loading with “some.thing” as a basename. */
+		const imageExtLoader_t *loader = R_FindCubeLoader( fileName );
+
+		/* Then if the image path is “something.crn” for a file named “something.crn”,
+		try loading with “something” as a basename. */
+		if ( *ext && !loader )
+		{
+			COM_StripExtension3( token, fileName, sizeof(fileName) );
```

Comment:
The `token` temporary pointer shouldn't be used after calling a bunch of other functions.